src/app/core/campaign/campaign.service.ts

```diff
@@ -38,38 +45,70 @@ export class CampaignService {
     if (!state.changed) {
       return of(state);
     } else if (state.remoteCampaign) {
-      return this.campaignDoc.update(state.localCampaign).pipe(
-        map(doc => {
-          this.campaignDoc = doc;
-          return this.docToCampaign(doc);
-        })
+      return this.campaignDoc$.pipe(
+        switchMap(doc => doc.update(state.localCampaign)),
+        map(doc => this.docToCampaignState(doc))
       );
     } else {
       return this.augury.root.pipe(
         switchMap(rootDoc => rootDoc.create('prx:campaign', {}, state.localCampaign)),
         map(doc => {
-          this.campaignDoc = doc;
-          return this.docToCampaign(doc);
+          return this.docToCampaignState(doc);
         })
       );
     }
   }
 
```

Comment:
loadCampaignZoomFlights basically does the same things as getFlights, but it returns Campaign and Flight models instead of State models. I want to keep State model awareness inside the store and have everything else just use the plain models. updateCampaign/createCampaign also duplicate functionality in putCampaign so we're just working with the plain models outside the store.

Reply:
+2 also a good call.  Encapsulate that state!